app/services/group/seed.rb

```diff
@@ -0,0 +1,53 @@
+module Group
+  class Seed
+    attr_reader :status
+
+    def initialize(tenant)
+      @user = ManageIQ::API::Common::Request.current.user
+      validate(tenant)
+    end
+
+    def validate(tenant)
+      account_number = ManageIQ::API::Common::Request.current.identity['identity']['account_number']
+      raise Catalog::NotAuthorized if account_number != tenant.external_tenant
+      raise Catalog::NotAuthorized unless @user.org_admin?
+
+      true
+    end
+
+    def tenant
+      @user.tenant
+    end
+
+    def code(status)
+      @status = status
+    end
+
+    def process
+      lookup_groups
```

Comment:
@syncrou should this happen only if the RbacSeed doesn't have the external tenant if it does then we should not be doing a lookup groups.